src/main/scala/ztis/model/ModelBuilder.scala

```diff
@@ -38,22 +39,48 @@ object ModelBuilder extends App with StrictLogging {
 
   logger.info(s"Count: ${training.count()}\n")
 
-  // TODO(#13)
-  val rank = 8
-  val lambda = 0.1
-  val numIters = 10
+  {
+    val ranks = config.getIntList("model.params.ranks").asScala.toVector.map(_.toInt)
+    val lambdas = config.getDoubleList("model.params.lambdas").asScala.toVector.map(_.toDouble)
+    val numIters = config.getIntList("model.params.iterations").asScala.toVector.map(_.toInt)
 
-  val model = ALS.train(training, rank, numIters, lambda)
+    val directory = s"report-${DateTime.now().toString(DateTimeFormat.forPattern("yyyy-MM-dd--HH-mm-ss"))}"
+    s"mkdir $directory" !!
 
-  val directory = s"report-${DateTime.now().toString(DateTimeFormat.forPattern("yyyy-MM-dd--HH-mm-ss"))}"
-  s"mkdir $directory" !!
+    val paramsCrossProduct = for {rank <- ranks; lambda <- lambdas; numIter <- numIters} yield (rank, lambda, numIter)
 
-  val score = evaluate(model, directory)
+    val results = paramsCrossProduct.map {
+      case params@(rank, lambda, numIter) => {
+        val modelDirectory = s"$directory/r$rank-l${(lambda * 100).toInt}-i$numIter"
+        s"mkdir $modelDirectory" !!
 
-  persistInDatabase(model)
+        val model = ALS.train(training, rank, numIter, lambda)
+        val score = evaluate(model, modelDirectory)
+
+        (score, model, params)
+      }
+    }.sortBy(-_._1).toList
+
+    val best = results.head
+    val (score, model, (rank, lambda, numIter)) = best
+
+    dumpScores(results, directory + "/scores.txt")
+
+    persistInDatabase(model)
+    logger.info(s"The best model params: rank=$rank, lambda=$lambda, numIter=$numIter. It's ROC AUC = $score")
+  }
 
   spark.stop()
 
+  def dumpScores(scores : List[(Double, MatrixFactorizationModel, (Int, Double, Int))], filename: String) = {
+    val header = "rank, lambda, numIter, score\n"
```

Comment:
For some CSV processing programs no spaces are better, so header would look like "rank,lambda,numIter,score\n"